disco/extensions/upgrade_simulation/upgrades/voltage_upgrade_functions.py

```
@@ -885,36 +924,111 @@ def determine_new_regulator_upgrades(voltage_config, buses_with_violations, volt
+                                                            voltage_upper_limit=voltage_upper_limit, voltage_lower_limit=voltage_lower_limit,
+                                                            exclude_sub_ltc=True, only_sub_ltc=False, **kwargs)
+            regcontrols_df, regcontrol_settings_commands_list = choose_best_regcontrol_sweep_setting(deciding_field=deciding_field,
+                regcontrol_sweep_df=regcontrol_sweep_df, initial_regcontrols_df=regcontrol_df, **kwargs)
+            reload_dss_circuit(dss_file_list=dss_file_list, commands_list=previous_dss_commands_list+temp+regcontrol_settings_commands_list, **kwargs)  # reload circuit after settings sweep
+            bus_voltages_df, undervoltage_bus_list, overvoltage_bus_list, buses_with_violations = get_bus_voltages(
+                voltage_upper_limit=voltage_upper_limit, voltage_lower_limit=voltage_lower_limit, **kwargs)
+            reg_upgrade_commands =  temp + regcontrol_settings_commands_list
+            if len(buses_with_violations) > 0:
+                regcontrol_df = get_regcontrol_info()
+                regcontrol_sweep_df = sweep_regcontrol_settings(voltage_config=new_voltage_config,
+                                                                initial_regcontrols_df=regcontrol_df,
+                                                                voltage_upper_limit=voltage_upper_limit, voltage_lower_limit=voltage_lower_limit,
+                                                                exclude_sub_ltc=False, only_sub_ltc=True, **kwargs)
+                regcontrols_df, subltc_settings_commands_list = choose_best_regcontrol_sweep_setting(deciding_field=deciding_field,
+                    regcontrol_sweep_df=regcontrol_sweep_df, initial_regcontrols_df=regcontrol_df, **kwargs)
+                reg_upgrade_commands =  reg_upgrade_commands + subltc_settings_commands_list
+                
+        except:

```

Same point here about catch-all exceptions

---

Added better exception handling